### PR TITLE
Add all signers to the Transaction when serializing with ToStackItem

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/2460

### CODE REVIEW

**Comment by cloud8little** [comment]
Testing

**Comment by cloud8little** [comment]
Since the Signers is converted to ByteString, how can it be converted back to Signer type?  The fourth result of Array type is the returned Signers.

```cs
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "script": "wh8MC2NoZWNrU2lnbmVyDBRTmj3IXRZ1Tf+gBpCEMh4YG3SmkEFifVtS",
    "state": "HALT",
    "gasconsumed": "1050210",
    "exception": null,
    "stack": [
      {
        "type": "Array",
        "value": [
          {
            "type": "ByteString",
            "value": "qO1jTXO3f137AmS91H3MSookn8NFTdPtqgaURWsdf68="
          },
          {
            "type": "Integer",
            "value": "0"
          },
          {
            "type": "Integer",
            "value": "0"
          },
          {
            "type": "Array",
            "value": [
              {
                "type": "ByteString",
                "value": "fu4aq+tn7R15HUTk9fzzrpFxqHEQBPlbZ8nQL4JCkov3isQhmpO9EP21fj0kMR6lYXRJjn3DJQem7XroYNUBXdiaWwTnWPtnw6I+BU6v20QSknqeDm1/DzmQMR8r4GiZ90DEZznR"
              },
              {
                "type": "ByteString",
                "value": "Edm1mjSxnQkQ+yCRGOv8WYg6rvwQBPlbZ8nQL4JCkov3isQhmpO9EP21fj0kMR6lYXRJjn3DJQem7XroYNUBXdiaWwTnWPtnw6I+BU6v20QSknqeDm1/DzmQMR8r4GiZ90DEZznR"
              }
            ]
          },
          {
            "type": "Integer",
            "value": "0"
          },
          {
            "type": "Integer",
            "value": "0"
          },
          {
            "type": "Integer",
            "value": "0"
          },
          {
            "type": "ByteString",
            "value": ""
          }
        ]
      }
    ]
  }
}
```

**Comment by superboyiii** [comment]
We should avoid any incompatible changes on RC3 only if it's very critical.

**Comment by shargon** [comment]
The option B it's serialize Signers at the end, I think that this option it's most optimal, what oyou prefer @erikzhang ?

**Comment by erikzhang** [comment]
> The option B it's serialize Signers at the end

I think it's okay. But why do we need this?

**Comment by shargon** [comment]
> I think it's okay. But why do we need this?

SC maybe will need to know all the signers

File: src/neo/Network/P2P/Payloads/Transaction.cs [hunk 1]
```diff
@@ -421,6 +421,7 @@ public StackItem ToStackItem(ReferenceCounter referenceCounter)
                 NetworkFee,
                 ValidUntilBlock,
                 Script,
+                new Array(referenceCounter, Signers.Select(u=> (ByteString)u.ToArray()))
```
Comment (erikzhang) [inline src/neo/Network/P2P/Payloads/Transaction.cs:424]: How to deserialize it in a contract?

Comment (shargon) [reply src/neo/Network/P2P/Payloads/Transaction.cs:424]: Only the address better?

Comment (erikzhang) [reply src/neo/Network/P2P/Payloads/Transaction.cs]: @csmuller Does it work for you?

Comment (erikzhang) [reply src/neo/Network/P2P/Payloads/Transaction.cs]: I think we should answer the question first: why do we need to read the signers in a contract?

Comment (roman-khimov) [reply src/neo/Network/P2P/Payloads/Transaction.cs]: To implement some checks in `verify`?

Comment (erikzhang) [reply src/neo/Network/P2P/Payloads/Transaction.cs]: Maybe we can create some syscalls to handle it? Just like `CheckWitness`?

**Pull request closed without merging**